packages/vite/client.d.ts

```diff
@@ -246,3 +246,20 @@ declare module '*?inline' {
   const src: string
   export default src
 }
+
+declare interface VitePreloadErrorEvent extends Error {
+  payload: Error
+}
+
+declare interface Window {
+  addEventListener(
+    type: 'vite:preloadError',
+    listener: (this: Window, ev: VitePreloadErrorEvent) => unknown,
+    options?: boolean | AddEventListenerOptions,
+  ): void
+  removeEventListener(
+    type: 'vite:preloadError',
+    listener: (this: Window, ev: VitePreloadErrorEvent) => unknown,
+    options?: boolean | AddEventListenerOptions,
```

Comment:
Shouldn't this be:

```suggestion
    options?: boolean | EventListenerOptions,
```

Referencing the TS `lib.dom.d.ts`:

```ts
interface Document {
    addEventListener<K extends keyof DocumentEventMap>(type: K, listener: (this: Document, ev: DocumentEventMap[K]) => any, options?: boolean | AddEventListenerOptions): void;
    addEventListener(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | AddEventListenerOptions): void;
    removeEventListener<K extends keyof DocumentEventMap>(type: K, listener: (this: Document, ev: DocumentEventMap[K]) => any, options?: boolean | EventListenerOptions): void;
    removeEventListener(type: string, listener: EventListenerOrEventListenerObject, options?: boolean | EventListenerOptions): void;
}
```

Reply:
It looks like `EventListenerOptions` only supports configuration of `capture`. However, I'm not sure whether it needs to support configuration of `once`, `passive`, and `signal`. 🤔

Reply:
I think those only make sense for `addEventListener`? It would probably be safer to follow what's defined by TypeScript.

Reply:
Ok.